frontend/util.lua

```diff
@@ -113,7 +113,13 @@ Source: <a href="https://gist.github.com/jesseadams/791673">https://gist.github.
 ---- @treturn string clock string in the form of 00:00 or 00:00:00
 function util.secondsToClock(seconds, withoutSeconds)
     seconds = tonumber(seconds)
-    if seconds == 0 or seconds ~= seconds then
+    if not seconds then
+        if withoutSeconds then
+            return "--:--"
+        else
+            return "--:--:--"
+        end
+    elseif seconds == 0 or seconds ~= seconds then
```

Comment:
Not your fault but how can `seconds ~= seconds` be true?

Reply:
I think whoever wrote it probably meant something like seconds == nil, since if you feed something into `tonumber()` that's not a number that's what you'll get (as now covered by `if not seconds` in this PR).

In any case, that `seconds ~= seconds` is indeed quite useless.

Reply:
Thank's a lot! I will have look at your links/work tomorrow morning. 
+1

Reply:
IIRC, the `LuaNumber ~= LuaNumber` idiom is used to check for NaNs. Not quite sure if it makes sense in this context, though ;).

Reply:
To remove or not to. That's the question.:)

Reply:
@NiLuJe 
> IIRC, the LuaNumber ~= LuaNumber idiom is used to check for NaNs. Not quite sure if it makes sense in this context, though ;).

You mean if you plug in something larger than 2^1024 or whatever the largest supported number is exactly?

Reply:
> @NiLuJe
> 
> > IIRC, the LuaNumber ~= LuaNumber idiom is used to check for NaNs. Not quite sure if it makes sense in this context, though ;).
> 
> You mean if you plug in something larger than 2^1024 or whatever the largest supported number is exactly?

Nope, an actual Lua NaN (i.e., "nan" or "#IND" on win, c.f., `print(0/0)`).

Reply:
Oh! If you do `tonumber()` on a nan (like you can generate with 0/0) it doesn't turn into nil like it would do for values that are, um, not numbers.

So while it doesn't *really* seem to serve a purpose, it's not useless in case you call something like `secondsToClock(0/0)`… somehow. >_>